pkg/credentialprovider/keyring.go

```diff
@@ -57,11 +57,11 @@ type lazyDockerKeyring struct {
 func (dk *BasicDockerKeyring) Add(cfg DockerConfig) {
 	if dk.index == nil {
 		dk.index = make([]string, 0)
-		dk.creds = make(map[string][]docker.AuthConfiguration)
+		dk.creds = make(map[string][]dockertypes.AuthConfig)
 	}
 	for loc, ident := range cfg {
 
-		creds := docker.AuthConfiguration{
+		creds := dockertypes.AuthConfig{
```

Comment:
@deads2k Thank you~ I think our `DockerConfig` is used to hold the content in `.dockercfg` or `.docker/config.json` (see #2409). Docker also uses the `AuthConfig` to hold the config file information (see [code](https://github.com/docker/docker/blob/4a7bd7eaef0038b006cf23adface84081e5aab3b/cliconfig/config.go#L161-L164)), so I think we could replace our own type with `AuthConfig` to cleanup the code and be align with docker implementation.